test/Microsoft.AspNetCore.Mvc.Core.Test/ModelBinding/Binders/KeyValuePairModelBinderTest.cs

```diff
@@ -125,7 +125,7 @@ public class KeyValuePairModelBinderTest
             var binder = new KeyValuePairModelBinder<int, string>(innerBinder, innerBinder);
 
             // Act
-            var result = await binder.TryBindStrongModel<int>(bindingContext, innerBinder, "key");
+            var result = await binder.TryBindStrongModel<int>(bindingContext, innerBinder, "Key");
```

Comment:
Yeah sorry missed this one. There's no trick. `.Properties[string]` is case-sensitive (as all things should be)